src/hooks/useWFSDownload.tsx

```diff
@@ -333,12 +343,21 @@ const useWFSDownload = (onCallback?: () => void) => {
       onCallback && onCallback();
 
       try {
+        // Extract download conditions
+        const dateRange = getDateConditionFrom(downloadConditions);
+        const multiPolygon = getMultiPolygonFrom(downloadConditions);
+        const format = getFormatFrom(downloadConditions); // TODO: currently unused - CSV only for WFS for now
+
+        // Set timeout for the entire download operation
+        const timeout = import.meta.env.VITE_WFS_DOWNLOADING_TIMEOUT;
+        const timeoutId = setTimeout(
+          () => controller.abort(),
+          Number(timeout) || CONNECTION_TIMEOUT_DEFAULT
+        );
+
+        // Start fetch with abort controller
         const controller = new AbortController();
         abortControllerRef.current = controller;
-        const timeout = process.env.VITE_WFS_DOWNLOADING_TIMEOUT
-          ? Number(process.env.VITE_WFS_DOWNLOADING_TIMEOUT)
-          : CONNECTION_TIMEOUT_DEFAULT; // Frontend 20 minutes timeout by default
-        const timeoutId = setTimeout(() => controller.abort(), timeout);
 
         const response = await fetch(
```

Comment:
I think it is better to push the ogc api call to the store as it is easier to find all these ogc call later, you can reference the fetchWithoutStore.. etc